### ISSUE TITLE
Doc.similarity GPU bug still present in 2.2.2: TypeError: Unsupported type <class 'numpy.ndarray'>

### ISSUE THREAD

**Issue by rjurney** [explosion/spaCy#4687]
## How to reproduce the behaviour

For full, working example using data from S3, see: https://github.com/rjurney/weakly_supervised_learning_code/blob/master/ch05/Snorkel.ipynb

```python
import cupy
import spacy

spacy.prefer_gpu()

# Download the spaCy english model
spacy.cli.download('en_core_web_lg')

nlp = spacy.load("en_core_web_lg")

doc1 = nlp(df['_Body'][0])
doc2 = nlp(df['_Body'][1])

print(type(doc1), type(doc2))
doc1.similarity(doc2)
```

Cell output / error:

```python
✔ Download and installation successful
You can now load the model via spacy.load('en_core_web_lg')
<class 'spacy.tokens.doc.Doc'> <class 'spacy.tokens.doc.Doc'>
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-319329f4354b> in <module>
     10 
     11 print(type(doc1), type(doc2))
---> 12 doc1.similarity(doc2)
     13 
     14 # df_sample['spacy'] = df['_Body'].apply(lambda x: nlp(x))

doc.pyx in spacy.tokens.doc.Doc.similarity()

doc.pyx in spacy.tokens.doc.Doc.vector_norm.__get__()

doc.pyx in __iter__()

cupy/core/core.pyx in cupy.core.core.ndarray.__add__()

cupy/core/_kernel.pyx in cupy.core._kernel.ufunc.__call__()

cupy/core/_kernel.pyx in cupy.core._kernel._preprocess_args()

TypeError: Unsupported type <class 'numpy.ndarray'>
```

Note that without `spacy.prefer_gpu()`, this works fine. Also note that cupy alone works fine. CUDA/cudNN setup is fine.

### Info about spaCy

* **spaCy version:** 2.2.2
* **Platform:** Linux-4.15.0-66-generic-x86_64-with-debian-buster-sid
* **Python version:** 3.7.4
* **CUDA version:** 10.0
* **CUDNN version:** 7.6.5.32-1+cuda10.0_amd64

* Operating System: Ubuntu Linux 18.04
* Python Version Used: Anaconda3-2019.07 Python 3.7.5
* Environment Information: pip 19.3.1 / pip-tools 4.2.0 / `pip-compile --output-file requirements.linux.txt requirements.in` / `pip install -r requirements.linux.txt`

Here is my `requirements.linux.txt`:

```python
#
# This file is autogenerated by pip-compile
# To update, run:
#
#    pip-compile --output-file=requirements.linux.txt requirements.in
#
-e git+git://github.com/snorkel-team/snorkel@master#egg=snorkel
absl-py==0.8.1            # via tensorboard, tensorflow-gpu
argh==0.26.2              # via watchdog
astor==0.8.0              # via tensorflow-gpu
attrs==19.3.0             # via jsonschema
backcall==0.1.0           # via ipython
beautifulsoup4==4.8.1
bleach==3.1.0             # via nbconvert
blessings==1.7            # via gpustat
blis==0.4.1               # via spacy, thinc
boto3==1.10.23
boto==2.49.0              # via smart-open
botocore==1.13.23         # via boto3, s3fs, s3transfer
cachetools==3.1.1         # via google-auth
certifi==2019.9.11        # via requests, sentry-sdk
chardet==3.0.4            # via requests
click==7.0                # via pip-tools, wandb
configparser==4.0.2       # via wandb
cupy-cuda100==6.5.0       # via spacy
cycler==0.10.0            # via matplotlib
cymem==2.0.3              # via preshed, spacy, thinc
decorator==4.4.1          # via ipython, networkx, traitlets
defusedxml==0.6.0         # via nbconvert
dill==0.3.1.1
docker-pycreds==0.4.0     # via wandb
docutils==0.15.2          # via botocore
entrypoints==0.3          # via nbconvert
fastrlock==0.4            # via cupy-cuda100
frozendict==1.2
fsspec==0.6.0             # via s3fs
gast==0.2.2               # via tensorflow-gpu
gensim==3.8.1
gitdb2==2.0.6             # via gitpython
gitpython==3.0.5          # via wandb
google-auth-oauthlib==0.4.1  # via tensorboard
google-auth==1.7.1        # via google-auth-oauthlib, tensorboard
google-pasta==0.1.8       # via tensorflow-gpu
gpustat==0.6.0
gql==0.1.0                # via wandb
graphql-core==2.2.1       # via gql
grpcio==1.25.0            # via tensorboard, tensorflow-gpu
h5py==2.10.0              # via keras-applications
idna==2.8                 # via requests
importlib-metadata==0.23  # via jsonschema, spacy
ipykernel==5.1.3          # via ipywidgets, jupyter, jupyter-console, notebook, qtconsole
ipython-genutils==0.2.0   # via nbformat, notebook, qtconsole, traitlets
ipython==7.9.0
ipywidgets==7.5.1         # via jupyter
iso8601==0.1.12
jedi==0.15.1              # via ipython
jinja2==2.10.3            # via nbconvert, notebook
jmespath==0.9.4           # via boto3, botocore
joblib==0.14.0            # via scikit-learn
jsonschema==3.2.0         # via nbformat
jupyter-client==5.3.4     # via ipykernel, jupyter-console, notebook, qtconsole
jupyter-console==6.0.0    # via jupyter
jupyter-core==4.6.1       # via jupyter-client, nbconvert, nbformat, notebook, qtconsole
jupyter==1.0.0
keras-applications==1.0.8  # via tensorflow-gpu
keras-preprocessing==1.1.0  # via tensorflow-gpu
kiwisolver==1.1.0         # via matplotlib
lxml==4.4.1
markdown==3.1.1           # via tensorboard
markupsafe==1.1.1         # via jinja2
matplotlib==3.1.1         # via seaborn
mistune==0.8.4            # via nbconvert
more-itertools==7.2.0     # via zipp
munkres==1.1.2
murmurhash==1.0.2         # via preshed, spacy, thinc
nbconvert==5.6.1          # via jupyter, notebook
nbformat==4.4.0           # via ipywidgets, nbconvert, notebook
networkx==2.3
nltk==3.4.5
notebook==6.0.2           # via jupyter, widgetsnbextension
numpy==1.17.4
nvidia-ml-py3==7.352.0    # via gpustat, wandb
oauthlib==3.1.0           # via requests-oauthlib
opt-einsum==3.1.0         # via tensorflow-gpu
pandas==0.25.3
pandocfilters==1.4.2      # via nbconvert
parso==0.5.1              # via jedi
pathtools==0.1.2          # via watchdog
pexpect==4.7.0            # via ipython
pickleshare==0.7.5        # via ipython
pip-tools==4.2.0
plac==1.1.3               # via spacy, thinc
preshed==3.0.2            # via spacy, thinc
prometheus-client==0.7.1  # via notebook
promise==2.2.1            # via gql, graphql-core
prompt-toolkit==2.0.10    # via ipython, jupyter-console
protobuf==3.10.0          # via tensorboard, tensorboardx, tensorflow-gpu, tensorflow-hub
psutil==5.6.5             # via gpustat, wandb
ptyprocess==0.6.0         # via pexpect, terminado
py4j==0.10.7              # via pyspark
pyarrow==0.14.1
pyasn1-modules==0.2.7     # via google-auth
pyasn1==0.4.8             # via pyasn1-modules, rsa
pygments==2.4.2           # via ipython, jupyter-console, nbconvert, qtconsole
pyparsing==2.4.5          # via matplotlib
pyrsistent==0.15.5        # via jsonschema
pyspark==2.4.4
python-dateutil==2.8.0    # via botocore, jupyter-client, matplotlib, pandas, wandb
pytz==2019.3              # via pandas
pyyaml==5.1.2             # via watchdog
pyzmq==18.1.1             # via jupyter-client, notebook
qtconsole==4.6.0          # via jupyter
requests-oauthlib==1.3.0  # via google-auth-oauthlib
requests==2.22.0
rsa==4.0                  # via google-auth
rx==1.6.1                 # via graphql-core
s3fs==0.4.0
s3transfer==0.2.1         # via boto3
scikit-learn==0.21.3
scipy==1.3.2              # via gensim, scikit-learn, seaborn
seaborn==0.9.0
send2trash==1.5.0         # via notebook
sentry-sdk==0.13.2        # via wandb
shortuuid==0.5.0          # via wandb
six==1.13.0               # via absl-py, bleach, blessings, cupy-cuda100, cycler, docker-pycreds, gensim, google-auth, google-pasta, gpustat, gql, graphql-core, grpcio, h5py, jsonschema, keras-preprocessing, nltk, pip-tools, promise, prompt-toolkit, protobuf, pyarrow, pyrsistent, python-dateutil, tensorboard, tensorboardx, tensorflow-gpu, tensorflow-hub, traitlets, wandb
smart-open==1.9.0         # via gensim
smmap2==2.0.5             # via gitdb2
soupsieve==1.9.5          # via beautifulsoup4
spacy[cuda100]==2.2.2
srsly==0.2.0              # via spacy, thinc
subprocess32==3.5.4       # via wandb
tensorboard==2.0.1        # via tensorflow-gpu
tensorboardx==1.9
tensorflow-estimator==2.0.1  # via tensorflow-gpu
tensorflow-gpu==2.0.0
tensorflow-hub==0.7.0
termcolor==1.1.0          # via tensorflow-gpu
terminado==0.8.3          # via notebook
testpath==0.4.4           # via nbconvert
textblob==0.15.3
texttable==1.6.2
thinc==7.3.1              # via spacy
torch==1.1.0
tornado==6.0.3            # via ipykernel, jupyter-client, notebook, terminado
tqdm==4.38.0              # via thinc
traitlets==4.3.3          # via ipykernel, ipython, ipywidgets, jupyter-client, jupyter-core, nbconvert, nbformat, notebook, qtconsole
urllib3==1.25.7           # via botocore, requests, sentry-sdk
wandb==0.8.15
wasabi==0.4.0             # via spacy, thinc
watchdog==0.9.0           # via wandb
wcwidth==0.1.7            # via prompt-toolkit
webencodings==0.5.1       # via bleach
werkzeug==0.16.0          # via tensorboard
wheel==0.33.6             # via tensorboard, tensorflow-gpu
widgetsnbextension==3.5.1  # via ipywidgets
wrapt==1.11.2             # via tensorflow-gpu
zipp==0.6.0               # via importlib-metadata

# The following packages are considered to be unsafe in a requirements file:
# setuptools==41.6.0        # via google-auth, ipython, jsonschema, kiwisolver, markdown, protobuf, spacy, tensorboard
```

Here is my `pip freeze` output:

```python
absl-py==0.8.1
argh==0.26.2
astor==0.8.0
attrs==19.3.0
backcall==0.1.0
beautifulsoup4==4.8.1
bleach==3.1.0
blessings==1.7
blis==0.4.1
boto==2.49.0
boto3==1.10.23
botocore==1.13.23
cachetools==3.1.1
certifi==2019.9.11
chardet==3.0.4
Click==7.0
configparser==4.0.2
cupy-cuda100==6.5.0
cycler==0.10.0
cymem==2.0.3
decorator==4.4.1
defusedxml==0.6.0
dill==0.3.1.1
docker-pycreds==0.4.0
docutils==0.15.2
entrypoints==0.3
fastrlock==0.4
frozendict==1.2
fsspec==0.6.0
gast==0.2.2
gensim==3.8.1
gitdb2==2.0.6
GitPython==3.0.5
google-auth==1.7.1
google-auth-oauthlib==0.4.1
google-pasta==0.1.8
gpustat==0.6.0
gql==0.1.0
graphql-core==2.2.1
grpcio==1.25.0
h5py==2.10.0
idna==2.8
importlib-metadata==0.23
ipykernel==5.1.3
ipython==7.9.0
ipython-genutils==0.2.0
ipywidgets==7.5.1
iso8601==0.1.12
jedi==0.15.1
Jinja2==2.10.3
jmespath==0.9.4
joblib==0.14.0
jsonschema==3.2.0
jupyter==1.0.0
jupyter-client==5.3.4
jupyter-console==6.0.0
jupyter-core==4.6.1
Keras-Applications==1.0.8
Keras-Preprocessing==1.1.0
kiwisolver==1.1.0
lxml==4.4.1
Markdown==3.1.1
MarkupSafe==1.1.1
matplotlib==3.1.1
mistune==0.8.4
more-itertools==7.2.0
munkres==1.1.2
murmurhash==1.0.2
nbconvert==5.6.1
nbformat==4.4.0
networkx==2.3
nltk==3.4.5
notebook==6.0.2
numpy==1.17.4
nvidia-ml-py3==7.352.0
oauthlib==3.1.0
opt-einsum==3.1.0
pandas==0.25.3
pandocfilters==1.4.2
parso==0.5.1
pathtools==0.1.2
pexpect==4.7.0
pickleshare==0.7.5
pip-tools==4.2.0
plac==1.1.3
preshed==3.0.2
prometheus-client==0.7.1
promise==2.2.1
prompt-toolkit==2.0.10
protobuf==3.10.0
psutil==5.6.5
ptyprocess==0.6.0
py4j==0.10.7
pyarrow==0.14.1
pyasn1==0.4.8
pyasn1-modules==0.2.7
Pygments==2.4.2
pyparsing==2.4.5
pyrsistent==0.15.5
pyspark==2.4.4
python-dateutil==2.8.0
pytz==2019.3
PyYAML==5.1.2
pyzmq==18.1.1
qtconsole==4.6.0
requests==2.22.0
requests-oauthlib==1.3.0
rsa==4.0
Rx==1.6.1
s3fs==0.4.0
s3transfer==0.2.1
scikit-learn==0.21.3
scipy==1.3.2
seaborn==0.9.0
Send2Trash==1.5.0
sentry-sdk==0.13.2
shortuuid==0.5.0
six==1.13.0
smart-open==1.9.0
smmap2==2.0.5
-e git://github.com/snorkel-team/snorkel@3cf96b797937c6a1d843e7e5dccd4ce8737c8a5a#egg=snorkel
soupsieve==1.9.5
spacy==2.2.2
srsly==0.2.0
subprocess32==3.5.4
tensorboard==2.0.1
tensorboardX==1.9
tensorflow-estimator==2.0.1
tensorflow-gpu==2.0.0
tensorflow-hub==0.7.0
termcolor==1.1.0
terminado==0.8.3
testpath==0.4.4
textblob==0.15.3
texttable==1.6.2
thinc==7.3.1
torch==1.1.0
tornado==6.0.3
tqdm==4.38.0
traitlets==4.3.3
urllib3==1.25.7
wandb==0.8.15
wasabi==0.4.0
watchdog==0.9.0
wcwidth==0.1.7
webencodings==0.5.1
Werkzeug==0.16.0
widgetsnbextension==3.5.1
wrapt==1.11.2
zipp==0.6.0
```

**Comment by adrianeboyd** [comment]
I suspect that this is the same bug as #4673 and that #4680 will fix this problem.

**Comment by adrianeboyd** [comment]
Fixed by #4680.
